### PR TITLE
Reuse previous request method when response status is 307.

### LINKED ISSUE #27: Use same request method when following 307 status
As per RFC:

> **307 Temporary Redirect**
>
> The 307 (Temporary Redirect) status code indicates that the target resource resides temporarily under a different URI and the user agent MUST NOT change the request method if it performs an automatic redirection to that URI.  Since the redirection can change over time, the client ought to continue using the original effective request URI for future requests.
> 
> The server SHOULD generate a Location header field in the response containing a URI reference for the different URI.  The user agent MAY use the Location field value for automatic redirection.  The server's response payload usually contains a short hypertext note with a hyperlink to the different URI(s).
>
> Note: This status code is similar to 302 (Found), except that it does not allow changing the request method from POST to GET.  This specification defines no equivalent counterpart for 301 (Moved Permanently) ([RFC7238], however, defines the status code 308 (Permanent Redirect) for this purpose).

Right now `GET` is [always being used](https://github.com/olalonde/follow-redirects/blob/master/create.js#L73-L74).

```js
function defaultMakeRequest(options, cb, res) {
  if (res) {
    // This is a redirect, so use only GET methods
    options.method = 'GET';
  }

  var req = options.nativeProtocol.request(options, cb);

  if (res) {
    // We leave the user to call `end` on the first request
    req.end();
  }

  return req;
}
```

### PR DESCRIPTION
**Pull request by fnando**
This PR fixes #27.

### CODE REVIEW

**Comment by leeyeh** [comment]
The same applies to 308.

**Comment by leeyeh** [comment]
@jamestalmage would you like to take a look?

File: .travis.yml [hunk 1]
```diff
@@ -1,14 +1,20 @@
 language: node_js
+sudo: false
+cache:
```
Comment (jamestalmage) [inline .travis.yml]: remove the caching lines

File: .travis.yml [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
+  directories:
+    - node_modules
 
 node_js:
-  - '0.12'
   - '0.10'
+  - '0.12'
   - '0.8'
+  - '4.3'
```
Comment (jamestalmage) [inline .travis.yml]: just use the major version:

```
- '4'
- '5'
```

Comment (jamestalmage) [inline .travis.yml]: add `6` as well.

remove `0.8` and `iojs`, they are completely EOL

**Comment by jamestalmage** [comment]
The issue I still wonder about is this:

What is supposed to happen to the payload when you redirect? It is especially problematic for streams. If you have already streamed some of the payload and get redirected, what do you do?

**Comment by fnando** [comment]
I'll apply your suggestions today. Thanks!

> What is supposed to happen to the payload when you redirect?

@jamestalmage The payload should be sent to the new uri, as per the RFC. That's how it works on both [httpie](http://httpie.org) and [curl](https://curl.haxx.se) btw.

**Comment by jamestalmage** [comment]
> The payload should be sent to the new uri, as per the RFC. That's how it works on both httpie and curl btw.

I think this creates a problem for stream bodies:

-- 

Per the spec on the [`100 Continue Response`](https://www.w3.org/Protocols/rfc2616/rfc2616-sec8.html#sec8.2.3):

> Therefore, when a client sends this header field to an origin server (possibly via a proxy) from which it has never seen a 100 (Continue) status, the client SHOULD NOT wait for an indefinite period before sending the request body

This is the way I read that: If you have never talked to that server before, and don't know if it will send a `100 (Continue)`, don't wait to stream the body- it might not support that part of the spec. Once you've seen a `100 Continue` from a server once, you can assume it does support `100 Continue`, and wait until you receive the `100` response to proceed with streaming the body.

--

Additionally, everything I read in the spec seems to indicate it would within spec for a server to send a `100 Continue` response, consume the body, and then send a `307` final response.

--

I don't think we need to handle the problem in this PR, but I suspect users are going to have problems with `307` redirects and streamed body payloads.

File: .travis.yml [hunk 3]
```diff
@@ -1,14 +1,10 @@
...
-  - '0.10'
-  - '0.8'
-  - 'iojs'
-
-before_install:
-- '[ "${TRAVIS_NODE_VERSION}" != "0.8" ] || npm install -g npm@1.4.28'
-- npm install -g npm@2.12
+  - '6'
+  - '5'
+  - '4'
```
Comment (jamestalmage) [inline .travis.yml]: Sorry if my earlier comment was unclear.

The list should be:

```
- '6'
- '5'
- '4'
- '0.12'
- '0.10'
```

Those are all the current supported versions of Node, which is what I'm trying to emulate.

**Comment by jamestalmage** [comment]
Closed in 3519070113ec9c0c8448c36ebc0c5959a7ae22a3

**Pull request closed without merging**